core/src/com/po/fuck/view/classdrawers/Misc.java

```diff
@@ -4,9 +4,33 @@
 import com.badlogic.gdx.graphics.g2d.Animation;
 import com.badlogic.gdx.graphics.g2d.TextureRegion;
 import com.badlogic.gdx.graphics.g2d.Sprite;
+import com.badlogic.gdx.utils.JsonValue;
+import com.po.fuck.Assets;
+import com.po.fuck.model.Core;
 
 public class Misc {
     public static Animation<TextureRegion> getAnimation(String internalPath){
-        return new Animation<TextureRegion>(1f, new Sprite(new Texture(internalPath)));
+        return new Animation<>(1f, new Sprite((Texture) Assets.manager.get(internalPath)));
+    }
+
+    public static Animation<TextureRegion> getComplixAnimation(String name) {
+        JsonValue info = Assets.getInfo(name);
+        Texture animatedTexture = Assets.manager.get(info.getString("path"));
+        int totalWidth = info.getInt("width");
+        int totalHeight = info.getInt("height");
+        int frameWidth = info.getInt("frameWidth");
+        int frameHeight = info.getInt("frameHeight");
+        TextureRegion[][] frames = TextureRegion.split(animatedTexture, frameWidth, frameHeight);
+        TextureRegion[] frames1 = new TextureRegion[(totalWidth/frameWidth) * (totalHeight/frameHeight)];
```

Comment:
Add some check for integer divisibility.

Reply:
What does `frames1` mean ?

Reply:
All frames, but in one dimension array

Reply:
@MAKMED1337 You want this check to be ```assert``` or what?